spec/models/user_profile_spec.rb

```diff
@@ -1,6 +1,43 @@
 # frozen_string_literal: true
 
-describe UserProfile do
+RSpec.describe UserProfile do
+  describe "Validations" do
+    subject(:profile) { user.user_profile }
+
+    fab!(:user) { Fabricate(:user) }
+    fab!(:watched_word) { Fabricate(:watched_word, word: "bad") }
+
+    describe "#location" do
+      context "when it contains watched words" do
+        before { profile.location = "bad location" }
+
+        it "is not valid" do
+          profile.valid?
+          expect(profile.errors[:base].size).to eq 1
```

Comment:
I think the assertion here needs to be more specific to check that the error is indeed that added by watched words. Otherwise, a future refactor may remove the watched words validator and add some other error to base and this assertion will still pass.